### ISSUE TITLE
Deploying specific functions does not work with version 13.0.2

### ISSUE THREAD

**Issue by jcurlier** [firebase/firebase-tools#6672]
### Environment info

**firebase-tools:**

13.0.2

**Platform:**

Mac, Ubuntu

### Steps to reproduce

```
npx firebase deploy --only functions:querySortedActivities --non-interactive
```

### [REQUIRED] Expected behavior

Working with version 12.9.1

```
=== Deploying to 'firebase-auratech16'...

i  deploying functions
i  functions: preparing codebase default for deployment
i  functions: ensuring required API cloudfunctions.googleapis.com is enabled...
i  functions: ensuring required API cloudbuild.googleapis.com is enabled...
i  artifactregistry: ensuring required API artifactregistry.googleapis.com is enabled...
✔  functions: required API cloudbuild.googleapis.com is enabled
✔  artifactregistry: required API artifactregistry.googleapis.com is enabled
✔  functions: required API cloudfunctions.googleapis.com is enabled
i  functions: Loading and analyzing source code for codebase default to determine what to deploy
Serving at port 8357

i  functions: preparing functions directory for uploading...
i  functions: packaged /Users/jeromecurlier/Development/Aura/Functions/FirebaseJob/functions (152.91 KB) for uploading
i  functions: ensuring required API cloudscheduler.googleapis.com is enabled...
✔  functions: required API cloudscheduler.googleapis.com is enabled
✔  functions: functions folder uploaded successfully
i  functions: updating Node.js 20 (1st Gen) function querySortedActivities(us-central1)...
✔  functions[querySortedActivities(us-central1)] Successful update operation.
i  functions: cleaning up build files...

✔  Deploy complete!

Project Console: https://console.firebase.google.com/project/firebase-auratech16/overview
```

### [REQUIRED] Actual behavior

Not working with version 13.0.2

```
Error: Cannot understand what targets to deploy/serve. No targets in firebase.json match '--only functions:querySortedActivities'.
```

**Comment by aalej** [comment]
Hey @jcurlier, thanks for reaching out. I think the cause of this could be related to our recent change in [#6555](https://github.com/firebase/firebase-tools/pull/6555). Could you share your `firebase.json` file so that we can check if this is the case? 

Also, does adding the following configuration to your `firebase.json` make any difference in the behavior:
```json
{
 "functions": [
   {
     "source": "functions",
     "codebase": "default",
     "ignore": [
       "node_modules",
       ".git",
       "firebase-debug.log",
       "firebase-debug.*.log"
     ]
   }
 ]
}
```

**Comment by jcurlier** [comment]
Yes this is the same issue - added "source": "functions" and solved it

**Comment by jcurlier** [comment]
Duplicate to https://github.com/firebase/firebase-tools/pull/6555.

**Comment by jcurlier** [comment]
my firebase.json was empty add adding the above solved the issue

**Comment by jcurlier** [comment]
This issue seems to be a duplicate of https://github.com/firebase/firebase-tools/pull/6555.

**Comment by aalej** [comment]
I'm glad to know that it's working now. Currently, this seems to be working as intended based on the changes in #6555, which was released in [v13.0.0](https://github.com/firebase/firebase-tools/releases/tag/v13.0.0) of `firebase-tools`. For now, I'm going to close this issue.
